# DAT210x - Programming with Python for DS

## Module6- Lab6

In [101]:
import pandas as pd
import time

### How to Get The Dataset

Grab the DLA HAR dataset from:

- http://groupware.les.inf.puc-rio.br/har
- http://groupware.les.inf.puc-rio.br/static/har/dataset-har-PUC-Rio-ugulino.zip
- A cached copy of the dataset is included in the course repository.

After extracting it out, load up the dataset into dataframe named `X` and do your regular dataframe examination:

In [102]:
# .. your code here ..
X = pd.read_csv('Datasets/dataset-har-PUC-Rio-ugulino.csv', sep=';')
X

user gender  age how_tall_in_meters  weight body_mass_index  \
0            debora  Woman   46               1,62      75            28,6   
1            debora  Woman   46               1,62      75            28,6   
2            debora  Woman   46               1,62      75            28,6   
3            debora  Woman   46               1,62      75            28,6   
4            debora  Woman   46               1,62      75            28,6   
5            debora  Woman   46               1,62      75            28,6   
6            debora  Woman   46               1,62      75            28,6   
7            debora  Woman   46               1,62      75            28,6   
8            debora  Woman   46               1,62      75            28,6   
9            debora  Woman   46               1,62      75            28,6   
10           debora  Woman   46               1,62      75            28,6   
11           debora  Woman   46               1,62      75            28,6   
12           debora  Woman   46               1,62      75            28,6   
13           debora  Woman   46               1,62      75            28,6   
14           debora  Woman   46               1,62      75            28,6   
15           debora  Woman   46               1,62      75            28,6   
16           debora  Woman   46               1,62      75            28,6   
17           debora  Woman   46               1,62      75            28,6   
18           debora  Woman   46               1,62      75            28,6   
19           debora  Woman   46               1,62      75            28,6   
20           debora  Woman   46               1,62      75            28,6   
21           debora  Woman   46               1,62      75            28,6   
22           debora  Woman   46               1,62      75            28,6   
23           debora  Woman   46               1,62      75            28,6   
24           debora  Woman   46               1,62      75            28,6   
25           debora  Woman   46               1,62      75            28,6   
26           debora  Woman   46               1,62      75            28,6   
27           debora  Woman   46               1,62      75            28,6   
28           debora  Woman   46               1,62      75            28,6   
29           debora  Woman   46               1,62      75            28,6   
...             ...    ...  ...                ...     ...             ...   
165603  jose_carlos    Man   75               1,67      67            24,0   
165604  jose_carlos    Man   75               1,67      67            24,0   
165605  jose_carlos    Man   75               1,67      67            24,0   
165606  jose_carlos    Man   75               1,67      67            24,0   
165607  jose_carlos    Man   75               1,67      67            24,0   
165608  jose_carlos    Man   75               1,67      67            24,0   
165609  jose_carlos    Man   75               1,67      67            24,0   
165610  jose_carlos    Man   75               1,67      67            24,0   
165611  jose_carlos    Man   75               1,67      67            24,0   
165612  jose_carlos    Man   75               1,67      67            24,0   
165613  jose_carlos    Man   75               1,67      67            24,0   
165614  jose_carlos    Man   75               1,67      67            24,0   
165615  jose_carlos    Man   75               1,67      67            24,0   
165616  jose_carlos    Man   75               1,67      67            24,0   
165617  jose_carlos    Man   75               1,67      67            24,0   
165618  jose_carlos    Man   75               1,67      67            24,0   
165619  jose_carlos    Man   75               1,67      67            24,0   
165620  jose_carlos    Man   75               1,67      67            24,0   
165621  jose_carlos    Man   75               1,67      67            24,0   
165622  jose_carlos    Man   75      

Encode the gender column such that: `0` is male, and `1` as female:

In [103]:
# .. your code here ..
X['gender'] = X.gender.map({'Woman':1, 'Man':0})
X

user  gender  age how_tall_in_meters  weight body_mass_index  \
0            debora       1   46               1,62      75            28,6   
1            debora       1   46               1,62      75            28,6   
2            debora       1   46               1,62      75            28,6   
3            debora       1   46               1,62      75            28,6   
4            debora       1   46               1,62      75            28,6   
5            debora       1   46               1,62      75            28,6   
6            debora       1   46               1,62      75            28,6   
7            debora       1   46               1,62      75            28,6   
8            debora       1   46               1,62      75            28,6   
9            debora       1   46               1,62      75            28,6   
10           debora       1   46               1,62      75            28,6   
11           debora       1   46               1,62      75            28,6   
12           debora       1   46               1,62      75            28,6   
13           debora       1   46               1,62      75            28,6   
14           debora       1   46               1,62      75            28,6   
15           debora       1   46               1,62      75            28,6   
16           debora       1   46               1,62      75            28,6   
17           debora       1   46               1,62      75            28,6   
18           debora       1   46               1,62      75            28,6   
19           debora       1   46               1,62      75            28,6   
20           debora       1   46               1,62      75            28,6   
21           debora       1   46               1,62      75            28,6   
22           debora       1   46               1,62      75            28,6   
23           debora       1   46               1,62      75            28,6   
24           debora       1   46               1,62      75            28,6   
25           debora       1   46               1,62      75            28,6   
26           debora       1   46               1,62      75            28,6   
27           debora       1   46               1,62      75            28,6   
28           debora       1   46               1,62      75            28,6   
29           debora       1   46               1,62      75            28,6   
...             ...     ...  ...                ...     ...             ...   
165603  jose_carlos       0   75               1,67      67            24,0   
165604  jose_carlos       0   75               1,67      67            24,0   
165605  jose_carlos       0   75               1,67      67            24,0   
165606  jose_carlos       0   75               1,67      67            24,0   
165607  jose_carlos       0   75               1,67      67            24,0   
165608  jose_carlos       0   75               1,67      67            24,0   
165609  jose_carlos       0   75               1,67      67            24,0   
165610  jose_carlos       0   75               1,67      67            24,0   
165611  jose_carlos       0   75               1,67      67            24,0   
165612  jose_carlos       0   75               1,67      67            24,0   
165613  jose_carlos       0   75               1,67      67            24,0   
165614  jose_carlos       0   75               1,67      67            24,0   
165615  jose_carlos       0   75               1,67      67            24,0   
165616  jose_carlos       0   75               1,67      67            24,0   
165617  jose_carlos       0   75               1,67      67            24,0   
165618  jose_carlos       0   75               1,67      67            24,0   
165619  jose_carlos       0   75               1,67      67            24,0   
165620  jose_carlos       0   75               1,67      67            24,0   
165621  jose_carlos       0   75               1,67      67      

Clean up any columns with commas in them so that they're properly represented as decimals:

In [104]:
# .. your code here ..
X['how_tall_in_meters'] = X.how_tall_in_meters.map(lambda s: s.replace(",","."))
X['body_mass_index'] = X.body_mass_index.map(lambda s: s.replace(",","."))
X

user  gender  age how_tall_in_meters  weight body_mass_index  \
0            debora       1   46               1.62      75            28.6   
1            debora       1   46               1.62      75            28.6   
2            debora       1   46               1.62      75            28.6   
3            debora       1   46               1.62      75            28.6   
4            debora       1   46               1.62      75            28.6   
5            debora       1   46               1.62      75            28.6   
6            debora       1   46               1.62      75            28.6   
7            debora       1   46               1.62      75            28.6   
8            debora       1   46               1.62      75            28.6   
9            debora       1   46               1.62      75            28.6   
10           debora       1   46               1.62      75            28.6   
11           debora       1   46               1.62      75            28.6   
12           debora       1   46               1.62      75            28.6   
13           debora       1   46               1.62      75            28.6   
14           debora       1   46               1.62      75            28.6   
15           debora       1   46               1.62      75            28.6   
16           debora       1   46               1.62      75            28.6   
17           debora       1   46               1.62      75            28.6   
18           debora       1   46               1.62      75            28.6   
19           debora       1   46               1.62      75            28.6   
20           debora       1   46               1.62      75            28.6   
21           debora       1   46               1.62      75            28.6   
22           debora       1   46               1.62      75            28.6   
23           debora       1   46               1.62      75            28.6   
24           debora       1   46               1.62      75            28.6   
25           debora       1   46               1.62      75            28.6   
26           debora       1   46               1.62      75            28.6   
27           debora       1   46               1.62      75            28.6   
28           debora       1   46               1.62      75            28.6   
29           debora       1   46               1.62      75            28.6   
...             ...     ...  ...                ...     ...             ...   
165603  jose_carlos       0   75               1.67      67            24.0   
165604  jose_carlos       0   75               1.67      67            24.0   
165605  jose_carlos       0   75               1.67      67            24.0   
165606  jose_carlos       0   75               1.67      67            24.0   
165607  jose_carlos       0   75               1.67      67            24.0   
165608  jose_carlos       0   75               1.67      67            24.0   
165609  jose_carlos       0   75               1.67      67            24.0   
165610  jose_carlos       0   75               1.67      67            24.0   
165611  jose_carlos       0   75               1.67      67            24.0   
165612  jose_carlos       0   75               1.67      67            24.0   
165613  jose_carlos       0   75               1.67      67            24.0   
165614  jose_carlos       0   75               1.67      67            24.0   
165615  jose_carlos       0   75               1.67      67            24.0   
165616  jose_carlos       0   75               1.67      67            24.0   
165617  jose_carlos       0   75               1.67      67            24.0   
165618  jose_carlos       0   75               1.67      67            24.0   
165619  jose_carlos       0   75               1.67      67            24.0   
165620  jose_carlos       0   75               1.67      67            24.0   
165621  jose_carlos       0   75               1.67      67      

Let's take a peek at your data types:

In [105]:
X.dtypes

user                  object
gender                 int64
age                    int64
how_tall_in_meters    object
weight                 int64
body_mass_index       object
x1                     int64
y1                     int64
z1                     int64
x2                     int64
y2                     int64
z2                     int64
x3                     int64
y3                     int64
z3                     int64
x4                     int64
y4                     int64
z4                     int64
class                 object
dtype: object

Convert any column that needs to be converted into numeric use `errors='raise'`. This will alert you if something ends up being problematic.

In [106]:
# .. your code here ..
X['how_tall_in_meters'] = pd.to_numeric(X['how_tall_in_meters'], errors='raise')
X['body_mass_index'] =pd.to_numeric(X['body_mass_index'], errors='raise')
X

user  gender  age  how_tall_in_meters  weight  body_mass_index  \
0            debora       1   46                1.62      75             28.6   
1            debora       1   46                1.62      75             28.6   
2            debora       1   46                1.62      75             28.6   
3            debora       1   46                1.62      75             28.6   
4            debora       1   46                1.62      75             28.6   
5            debora       1   46                1.62      75             28.6   
6            debora       1   46                1.62      75             28.6   
7            debora       1   46                1.62      75             28.6   
8            debora       1   46                1.62      75             28.6   
9            debora       1   46                1.62      75             28.6   
10           debora       1   46                1.62      75             28.6   
11           debora       1   46                1.62      75             28.6   
12           debora       1   46                1.62      75             28.6   
13           debora       1   46                1.62      75             28.6   
14           debora       1   46                1.62      75             28.6   
15           debora       1   46                1.62      75             28.6   
16           debora       1   46                1.62      75             28.6   
17           debora       1   46                1.62      75             28.6   
18           debora       1   46                1.62      75             28.6   
19           debora       1   46                1.62      75             28.6   
20           debora       1   46                1.62      75             28.6   
21           debora       1   46                1.62      75             28.6   
22           debora       1   46                1.62      75             28.6   
23           debora       1   46                1.62      75             28.6   
24           debora       1   46                1.62      75             28.6   
25           debora       1   46                1.62      75             28.6   
26           debora       1   46                1.62      75             28.6   
27           debora       1   46                1.62      75             28.6   
28           debora       1   46                1.62      75             28.6   
29           debora       1   46                1.62      75             28.6   
...             ...     ...  ...                 ...     ...              ...   
165603  jose_carlos       0   75                1.67      67             24.0   
165604  jose_carlos       0   75                1.67      67             24.0   
165605  jose_carlos       0   75                1.67      67             24.0   
165606  jose_carlos       0   75                1.67      67             24.0   
165607  jose_carlos       0   75                1.67      67             24.0   
165608  jose_carlos       0   75                1.67      67             24.0   
165609  jose_carlos       0   75                1.67      67             24.0   
165610  jose_carlos       0   75                1.67      67             24.0   
165611  jose_carlos       0   75                1.67      67             24.0   
165612  jose_carlos       0   75                1.67      67             24.0   
165613  jose_carlos       0   75                1.67      67             24.0   
165614  jose_carlos       0   75                1.67      67             24.0   
165615  jose_carlos       0   75                1.67      67             24.0   
165616  jose_carlos       0   75                1.67      67             24.0   
165617  jose_carlos       0   75                1.67      67             24.0   
165618  jose_carlos       0   75                1.67      67             24.0   
165619  jose_carlos       0   75                1.67      67             24.0   
165620  jose_carlos       0   75              

In [107]:
X = X.drop_duplicates()
 

In [108]:
X.shape

(164281, 19)

If you find any problematic records, drop them before calling the `to_numeric` methods above.

Okay, now encode your `y` value as a Pandas dummies version of your dataset's `class` column:

In [109]:
# .. your code here ..
X = pd.get_dummies(X, columns=['class'])
X

user  gender  age  how_tall_in_meters  weight  body_mass_index  \
0            debora       1   46                1.62      75             28.6   
1            debora       1   46                1.62      75             28.6   
2            debora       1   46                1.62      75             28.6   
3            debora       1   46                1.62      75             28.6   
4            debora       1   46                1.62      75             28.6   
5            debora       1   46                1.62      75             28.6   
6            debora       1   46                1.62      75             28.6   
7            debora       1   46                1.62      75             28.6   
8            debora       1   46                1.62      75             28.6   
9            debora       1   46                1.62      75             28.6   
10           debora       1   46                1.62      75             28.6   
11           debora       1   46                1.62      75             28.6   
12           debora       1   46                1.62      75             28.6   
13           debora       1   46                1.62      75             28.6   
14           debora       1   46                1.62      75             28.6   
15           debora       1   46                1.62      75             28.6   
16           debora       1   46                1.62      75             28.6   
17           debora       1   46                1.62      75             28.6   
18           debora       1   46                1.62      75             28.6   
19           debora       1   46                1.62      75             28.6   
20           debora       1   46                1.62      75             28.6   
21           debora       1   46                1.62      75             28.6   
22           debora       1   46                1.62      75             28.6   
23           debora       1   46                1.62      75             28.6   
24           debora       1   46                1.62      75             28.6   
25           debora       1   46                1.62      75             28.6   
26           debora       1   46                1.62      75             28.6   
27           debora       1   46                1.62      75             28.6   
28           debora       1   46                1.62      75             28.6   
29           debora       1   46                1.62      75             28.6   
...             ...     ...  ...                 ...     ...              ...   
165603  jose_carlos       0   75                1.67      67             24.0   
165604  jose_carlos       0   75                1.67      67             24.0   
165605  jose_carlos       0   75                1.67      67             24.0   
165606  jose_carlos       0   75                1.67      67             24.0   
165607  jose_carlos       0   75                1.67      67             24.0   
165608  jose_carlos       0   75                1.67      67             24.0   
165609  jose_carlos       0   75                1.67      67             24.0   
165610  jose_carlos       0   75                1.67      67             24.0   
165611  jose_carlos       0   75                1.67      67             24.0   
165612  jose_carlos       0   75                1.67      67             24.0   
165613  jose_carlos       0   75                1.67      67             24.0   
165614  jose_carlos       0   75                1.67      67             24.0   
165615  jose_carlos       0   75                1.67      67             24.0   
165616  jose_carlos       0   75                1.67      67             24.0   
165617  jose_carlos       0   75                1.67      67             24.0   
165618  jose_carlos       0   75                1.67      67             24.0   
165619  jose_carlos       0   75                1.67      67             24.0   
165620  jose_carlos       0   75              

In fact, get rid of the `user` and `class` columns:

In [110]:
# .. your code here ..
y = X[['class_sitting', 'class_sittingdown', 'class_standing', 'class_standingup', 'class_walking']]
X= X.drop(['user', 'class_sitting', 'class_sittingdown', 'class_standing', 'class_standingup', 'class_walking'], axis=1)
X

gender  age  how_tall_in_meters  weight  body_mass_index  x1   y1  \
0            1   46                1.62      75             28.6  -3   92   
1            1   46                1.62      75             28.6  -3   94   
2            1   46                1.62      75             28.6  -1   97   
3            1   46                1.62      75             28.6  -2   96   
4            1   46                1.62      75             28.6  -1   96   
5            1   46                1.62      75             28.6  -2   95   
6            1   46                1.62      75             28.6   1  100   
7            1   46                1.62      75             28.6  -1   97   
8            1   46                1.62      75             28.6  -1   98   
9            1   46                1.62      75             28.6   0   98   
10           1   46                1.62      75             28.6  -1   95   
11           1   46                1.62      75             28.6   3  100   
12           1   46                1.62      75             28.6   0   96   
13           1   46                1.62      75             28.6   0   97   
14           1   46                1.62      75             28.6   0   97   
15           1   46                1.62      75             28.6  -1   96   
16           1   46                1.62      75             28.6   2   99   
17           1   46                1.62      75             28.6   0   96   
18           1   46                1.62      75             28.6   0   97   
19           1   46                1.62      75             28.6   0   98   
20           1   46                1.62      75             28.6  -1   95   
21           1   46                1.62      75             28.6   1   99   
22           1   46                1.62      75             28.6  -3   96   
23           1   46                1.62      75             28.6  -1   97   
24           1   46                1.62      75             28.6   1   97   
25           1   46                1.62      75             28.6  -3   96   
26           1   46                1.62      75             28.6   3  100   
27           1   46                1.62      75             28.6  -4   95   
28           1   46                1.62      75             28.6   1  100   
29           1   46                1.62      75             28.6   0   96   
...        ...  ...                 ...     ...              ...  ..  ...   
165603       0   75                1.67      67             24.0 -39  147   
165604       0   75                1.67      67             24.0   7  109   
165605       0   75                1.67      67             24.0 -18  110   
165606       0   75                1.67      67             24.0   3  112   
165607       0   75                1.67      67             24.0  -3   69   
165608       0   75                1.67      67             24.0 -35   71   
165609       0   75                1.67      67             24.0 -26  120   
165610       0   75                1.67      67             24.0  13   66   
165611       0   75                1.67      67             24.0 -22  106   
165612       0   75                1.67      67             24.0   8  114   
165613       0   75                1.67      67             24.0  -7   99   
165614       0   75                1.67      67             24.0 -14  117   
165615       0   75                1.67      67             24.0  -6  103   
165616       0   75                1.67      67             24.0 -17   84   
165617       0   75                1.67      67             24.0 -11   87   
165618       0   75                1.67      67             24.0  -4   88   
165619       0   75                1.67      67             24.0  -8   85   
165620       0   75                1.67      67             24.0   1   92   
165621       0   75                1.67      67             24.0  -1   91   
165622       0   75                1.67      67             24.0  -2   94   
1656

Let's take a look at your handy-work:

In [111]:
X.describe()

gender            age  how_tall_in_meters         weight  \
count  164281.000000  164281.000000       164281.000000  164281.000000   
mean        0.613796      38.276392            1.639447      70.754056   
std         0.486880      13.214387            0.052788      11.305053   
min         0.000000      28.000000            1.580000      55.000000   
25%         0.000000      28.000000            1.580000      55.000000   
50%         1.000000      31.000000            1.620000      75.000000   
75%         1.000000      46.000000            1.710000      83.000000   
max         1.000000      75.000000            1.710000      83.000000   

       body_mass_index             x1             y1             z1  \
count    164281.000000  164281.000000  164281.000000  164281.000000   
mean         26.172411      -6.704494      88.429210     -93.496594   
std           3.000181      11.629704      23.922889      39.363166   
min          22.000000    -306.000000    -271.000000    -603.000000   
25%          22.000000     -12.000000      79.000000    -120.000000   
50%          28.400000      -6.000000      94.000000     -98.000000   
75%          28.600000       0.000000     101.000000     -64.000000   
max          28.600000     509.000000     533.000000     411.000000   

                  x2             y2             z2             x3  \
count  164281.000000  164281.000000  164281.000000  164281.000000   
mean      -88.379807     -52.476702    -176.200656      17.212532   
std       169.987931     205.898932     193.128290      52.719202   
min      -494.000000    -517.000000    -617.000000    -499.000000   
25%       -35.000000     -29.000000    -142.000000       9.000000   
50%        -9.000000      28.000000    -119.000000      22.000000   
75%         4.000000      86.000000     -29.000000      34.000000   
max       473.000000     295.000000     122.000000     507.000000   

                  y3             z3             x4             y4  \
count  164281.000000  164281.000000  164281.000000  164281.000000   
mean      104.742855     -93.888812    -168.040869     -92.435869   
std        54.291702      45.541792      38.112276      19.906949   
min      -506.000000    -613.000000    -702.000000    -526.000000   
25%        95.000000    -103.000000    -190.000000    -102.000000   
50%       107.000000     -90.000000    -169.000000     -91.000000   
75%       120.000000     -80.000000    -154.000000     -80.000000   
max       517.000000     410.000000     -13.000000      86.000000   

                  z4  
count  164281.000000  
mean     -159.589867  
std        13.226463  
min      -537.000000  
25%      -167.000000  
50%      -160.000000  
75%      -153.000000  
max       -43.000000

You can also easily display which rows have nans in them, if any:

In [112]:
X[pd.isnull(X).any(axis=1)]

Empty DataFrame
Columns: [gender, age, how_tall_in_meters, weight, body_mass_index, x1, y1, z1, x2, y2, z2, x3, y3, z3, x4, y4, z4]
Index: []

Create an RForest classifier named `model` and set `n_estimators=30`, the `max_depth` to 10, `oob_score=True`, and `random_state=0`:

In [113]:
# .. your code here ..
from sklearn.ensemble import RandomForestClassifier
model = RandomForestClassifier(bootstrap=True, n_estimators=30, max_depth=10, oob_score=True, random_state=0)

Split your data into `test` / `train` sets. Your `test` size can be 30%, with `random_state` 7. Use variable names: `X_train`, `X_test`, `y_train`, and `y_test`:

In [114]:
# .. your code here ..
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=7)

In [115]:
y_test

class_sitting  class_sittingdown  class_standing  class_standingup  \
70880               0                  0               1                 0   
5650                1                  0               0                 0   
148377              0                  0               0                 0   
57991               0                  1               0                 0   
90455               0                  0               1                 0   
44532               1                  0               0                 0   
63967               0                  0               1                 0   
26496               1                  0               0                 0   
24326               1                  0               0                 0   
144613              0                  0               0                 0   
3739                1                  0               0                 0   
94486               0                  0               1                 0   
20853               1                  0               0                 0   
136164              0                  0               0                 0   
125107              0                  0               0                 0   
72608               0                  0               1                 0   
13493               1                  0               0                 0   
61745               0                  1               0                 0   
49703               1                  0               0                 0   
465                 1                  0               0                 0   
126728              0                  0               0                 0   
122670              0                  0               0                 0   
86184               0                  0               1                 0   
27815               1                  0               0                 0   
93841               0                  0               1                 0   
157323              0                  0               0                 0   
133183              0                  0               0                 0   
124485              0                  0               0                 0   
113436              0                  0               0                 1   
107680              0                  0               1                 0   
...               ...                ...             ...               ...   
161584              0                  0               0                 0   
62353               0                  1               0                 0   
44068               1                  0               0                 0   
82792               0                  0               1                 0   
121964              0                  0               0                 1   
78420               0                  0               1                 0   
11232               1                  0               0                 0   
45999               1                  0               0                 0   
93533               0                  0               1                 0   
163219              0                  0               0                 0   
137881              0                  0               0                 0   
58507               0                  1               0                 0   
13258               1                  0               0                 0   
136405              0                  0               0                 0   
19945               1                  0               0                 0   
2278                1                  0               0                 0   
129265              0                  0               0                 0   
150542              0                  0               0                 0   
17479               1                  0               0                 0   
122326              0         

### Now the Fun Stuff

In [116]:
print("Fitting...")
s = time.time()

# TODO: train your model on your training set

# .. your code here ..

model.fit(X_train,y_train) 
#model.fit(X,y)
print("Fitting completed in: ", time.time() - s)

Fitting...
Fitting completed in:  8.306702852249146


Display the OOB Score of your data:

In [117]:
score = model.oob_score_
print("OOB Score: ", round(score*100, 3))

OOB Score:  98.742


In [118]:
print("Scoring...")
s = time.time()

# TODO: score your model on your test set

# .. your code here ..
score = model.score(X_test,y_test)
print("Score: ", round(score*100, 3))
print("Scoring completed in: ", time.time() - s)

Scoring...
Score:  95.999
Scoring completed in:  0.8438591957092285


At this point, go ahead and answer the lab questions, then return here to experiment more --

Try playing around with the gender column. For example, encode `gender` `Male:1`, and `Female:0`. Also try encoding it as a Pandas dummies variable and seeing what changes that has. You can also try dropping gender entirely from the dataframe. How does that change the score of the model? This will be a key insight on how your feature encoding alters your overall scoring, and why it's important to choose good ones.

In [119]:
# .. your code changes above ..

After that, try messing with `y`. Right now its encoded with dummies, but try other encoding methods to what effects they have.

Please enter a numeric value (e.g. 0, 1, 10.5, etc) which correctly answers the question(s) below:

Whichever score ended up being the lesser value, either the OOB score or the model's accuracy score, enter that figure below:

95.999
incorrect95.687
95.999 
Submit You have used 2 of 2 attempts Some problems have options such as save, reset, hints, or show answer. These options follow the Submit button.